.github/workflows/lint.yml

```
@@ -0,0 +1,70 @@
+# Copyright 2022 ETH Zurich and University of Bologna.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+
+# Author: Sergio Mazzola <smazzola@iis.ee.ethz.ch>
+
+name: lint
+
+on: [ push, pull_request, workflow_dispatch ]
```

This is a quicker action that might make sense to run on all commits (but two runs of the same/similar are not needed imo, even though the pull_request trigger may run on slightly different code).
```suggestion
on: [ push, workflow_dispatch ]
```
Happy to discuss though if you think a different approach makes more sense.